### PR TITLE
Implement add/remove service endpoint methods.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
Ready for review.

### CODE REVIEW

**Review by mattcollier — APPROVED** [review]

File: lib/methods/veres-one/veres-did-doc.js [hunk 1]
```diff
@@ -240,6 +240,106 @@ class VeresOneDidDoc {
...
+   * Adds a service endpoint to this did doc.
+   * One of `id` or `name` is required.
+   *
+   * @param [id] {string}
+   * @param [name] {string}
+   * @param type {string} Endpoint type (e.g. 'AgentService')
+   * @param endpoint {string} Endpoint uri (e.g. 'https://agent.example.com')
+   * @param [options] {object} Any additional properties of endpoint
+   */
+  addService({id, name, type, endpoint, ...options}) {
```
Comment (davidlehn) [inline lib/methods/veres-one/veres-did-doc.js:265]: This API is a bit odd.  The docs above imply you pass in an optional options object, but the code is just slurping in all extra params into options.  That design does make it impossible to include `name` or `endpoint` in the endpoint object.  And you could cause madness by passing in different `endpoint` and `serviceEndpoint` values.  I'm not sure what to suggest here, but at least clear up the docs.

Comment (dmitrizagidulin) [reply lib/methods/veres-one/veres-did-doc.js:265]: Apologies, I think the problem is just in the jsdoc (I wasn't sure how to denote `...options` in its syntax). 
The idea I was going for is - you should be able to pass in arbitrary properties for the node, yeah?

Comment (davidlehn) [reply lib/methods/veres-one/veres-did-doc.js:265]: Yeah. I'm not sure how to do that in jsdoc.  The edge case of using extra properties named `endpoint` or `name` could be solved with an actual options object.  Anyone else have thoughts on what this API should look like and allow?

Comment (dmitrizagidulin) [reply lib/methods/veres-one/veres-did-doc.js:265]: Open to suggestions on this (or we can open an issue to revisit it..?)

File: lib/methods/veres-one/veres-did-doc.js [hunk 2]
```diff
@@ -206,6 +206,106 @@ class VeresOneDidDoc {
...
+   * @param serviceName {string}
+   *
+   * @returns {string} Service id
+   */
+  serviceIdFor(serviceName) {
+    if(!serviceName) {
+      throw new Error('Invalid service name');
+    }
+
+    return this.id + ';' + encodeURIComponent(serviceName);
```
Comment (dlongley) [inline lib/methods/veres-one/veres-did-doc.js]: This is ok for now, but please file an issue about how we may be quickly revisiting it to be something like:

```js
return this.id + ';service=' + encodeURIComponent(serviceName);`
```

File: lib/methods/veres-one/veres-did-doc.js [hunk 3]
```diff
@@ -206,6 +206,106 @@ class VeresOneDidDoc {
...
+    const serviceId = id || this.serviceIdFor(name);
+
+    if(!type) {
+      throw new Error('Service endpoint type is required');
+    }
+    if(!endpoint) {
+      throw new Error('Service endpoint uri is required');
+    }
+
+    jsonld.addValue(this.doc, 'service', {
```
Comment (dlongley) [inline lib/methods/veres-one/veres-did-doc.js:281]: Do we need to be more precise here and augment/replace a service or throw an error if it already exists? I would think so.

**Review by dlongley — CHANGES_REQUESTED** [review]
LGTM other than dealing with duplicate services ... I think we need to be more robust there.

File: lib/methods/veres-one/veres-did-doc.js (hunk 3 above)
Comment (dlongley) [reply lib/methods/veres-one/veres-did-doc.js:281]: The quickest fix is to throw an error for a duplicate service, forcing a service to be removed prior to adding it with changes.

File: lib/methods/veres-one/veres-did-doc.js (hunk 2 above)
Comment (dmitrizagidulin) [reply lib/methods/veres-one/veres-did-doc.js]: Well, I mean, I can just change it to that right now :) s' not a very big change.

File: lib/methods/veres-one/veres-did-doc.js (hunk 3 above)
Comment (dmitrizagidulin) [reply lib/methods/veres-one/veres-did-doc.js:281]: Ah, I think the context was - did-client was doing a check for an existing service downstream, in client code. So I didn't include it in the 'addService()'. But that's probably a better place for it, yeah.

Comment (dmitrizagidulin) [reply lib/methods/veres-one/veres-did-doc.js:281]: K, should be fixed in latest commit.

**Pull request merged**